src/Compilers/CSharp/Portable/Symbols/Extensions/SourceExtensionImplementationMethodSymbol.cs

```diff
@@ -78,10 +81,38 @@ internal sealed override void AddSynthesizedAttributes(PEModuleBuilder moduleBui
                 }
             }
 
-            base.AddSynthesizedAttributes(moduleBuilder, ref attributes);
             SourceMethodSymbol.AddSynthesizedAttributes(this, moduleBuilder, ref attributes);
         }
 
+        internal override void AddSynthesizedReturnTypeAttributes(PEModuleBuilder moduleBuilder, ref ArrayBuilder<CSharpAttributeData> attributes)
```

Comment:
>AddSynthesizedReturnTypeAttributes [](http://example.com/codeflow?start=31&length=34)

The logic in this method is very different from the logic in ```SourcePropertyAccessorSymbol.AddSynthesizedReturnTypeAttributes```. It is quite possible that it achieves the same thing, but it is hard to tell. At the same time, I think our intent is to achieve exactly the same thing. Consider extracting a helper that deals with synthesizing the attributes for ```SourcePropertyAccessorSymbol``` into a helper in that class and simply calling it on ```_originalMethod``` from here when appropriate.